docs/dev/guidelines/server.rst

```diff
@@ -431,25 +429,19 @@ For more details, please visit the :doc:`./criteria-builder` page.
 22. REST endpoint best practices for authorization
 ==================================================
 
-To reject unauthorized requests as early as possible, Artemis employs a two-step system:
-
-#. ``PreAuthorize`` and ``Enforce`` annotations are responsible for blocking users with wrong or missing authorization roles without querying the database.
-#. The ``AuthorizationCheckService`` is responsible for checking access rights to individual resources by querying the database.
+To reject unauthorized requests as early as possible, Artemis employs two solutions:
 
-Because the first method without database queries is substantially faster, always annotate your REST endpoints with the corresponding annotation. Always use the annotation for the minimum role that has access.
-The following example makes the call only accessible to ADMIN and INSTRUCTOR users:
-
-.. code-block:: java
+#. Implicit pre- and post-authorization annotations:
+    #. ``EnforceRoleInResource`` (e.g. ``EnforceAtLeastInstructorInCourse``) annotations are responsible for blocking users with wrong or missing authorization roles without querying the database.
+    #. If necessary, these annotations check for access rights to individual resources within the database via light-weight queries.
```

Comment:
The first line states that there is no querying and the second one states that there might be light-weight queries. This seams a bit confusing...